py4DSTEM/process/diffraction/crystal.py

```diff
@@ -400,6 +410,149 @@ def from_pymatgen_structure(
 
         return Crystal(positions, numbers, cell)
 
+    def from_ase(
+        atoms,
+    ):
+        """
+        Create a py4DSTEM Crystal object from an ASE atoms object
+
+        Args:
+            atoms (ase.Atoms): an ASE atoms object
+
+        """
+
+        xtal = Crystal(
+            positions=atoms.get_scaled_positions(),  # fractional coords
+            numbers=atoms.numbers,
+            cell=atoms.cell.array,
+        )
+
+        # get occupancies
+        # ASE seems to have different ways of storing occupancies
+        # if ASE object created from prismatic file
+        if "occupancies" in atoms.arrays.keys():
+            # np.array with length number of atoms
+            xtal.occupancy = atoms.arrays["occupancies"]
+            # if created from cif file
+        elif "occupancy" in atoms.info.keys():
+            # python dict with occupancy per site
+            xtal.occupancy = atoms.info["occupancy"]
+        # TODO add in elif statement if other ways appear
+        else:
+            print(Warning("Could not find occupancies of crystal"))
+        return xtal
+
+    def from_prismatic(filepath):
+        """
+        Create a py4DSTEM Crystal object from an prismatic style xyz co-ordinate file
+
+        Args:
+            filepath (str|Pathlib.Path): path to the prismatic format xyz file
+
+        """
+
+        # check if ase is installed
+        if find_spec("ase") is None:
+            raise ImportWarning("Could not import ASE, please install, and try again")
+        else:
+            from ase.io import read
+
+        atoms = read(filepath, format="prismatic")
+
+        # create the crystal object
+        xtal = Crystal(
+            positions=atoms.get_scaled_positions(),  # fractional coords
+            numbers=atoms.numbers,
+            cell=atoms.cell.array,
+        )
+        # add occupancies
+        # It should be this one but keeping other method in case
+        if "occupancies" in atoms.arrays.keys():
+            # np.array with length number of atoms
+            xtal.occupancy = atoms.arrays["occupancies"]
+            # if created from cif file
+        elif "occupancy" in atoms.info.keys():
+            # python dict with occupancy per site
+            xtal.occupancy = atoms.info["occupancy"]
+        # TODO add in elif statement if other ways appear
+        else:
+            print(Warning("Could not find occupancies of crystal"))
+        return xtal
+
+    def from_cif(
+        filepath,
+    ):
+        """
+        Create a py4DSTEM Crystal object from a cif file using ase.io.read function
+
+        Args:
+            filepath (str|Pathlib.Path): path to the file
+        """
+
+        # check if ase is installed
+        if find_spec("ase") is None:
+            raise ImportWarning(
+                "Could not import ASE, please install, restart and try again"
+            )
+        else:
+            from ase.io import read
```

Comment:
Is all this extra logic necessary? If `ase` is missing it will raise a ModuleNotFoundError

Reply:
I'm going to delete as its now a dependency